### ISSUE TITLE
Lint/LiteralInInterpolation does incorrect replacement for hashes

### ISSUE THREAD

**Issue by tristil** [rubocop/rubocop#11076]
Running the autocorrect for Lint/LiteralInInterpolation does the right thing for strings and numbers, but it does an incorrect replacement for hashes. This is because unlike strings and numbers the string representation of a hash is different from how it is written in code. 

--------

## Expected behavior

```ruby
# original code
"#{ { foo: 'bar'} }" #=> "{:foo=>\"bar\"}"

# autocorrected code
"{:foo=>\"bar\"}"
```

Alternatively, and maybe more preferably, this cop could just not complain about hash literals since depending on the string representation of a hash is an unusual use case.

## Actual behavior

```ruby
# original code
"#{ { foo: 'bar'} }" #=> "{:foo=>\"bar\"}"

# autocorrected code
"{ foo: 'bar' }"
```

## Steps to reproduce the problem

1. Create a Ruby file `foo.rb` with contents `"#{ { foo: 'bar'} }" `
2. run `rubocop --autocorrect --only Lint/LiteralInInterpolation foo.rb`

## RuboCop version

```
$ [bundle exec] rubocop -V
1.36.0 (using Parser 3.1.2.1, rubocop-ast 1.21.0, running on ruby 2.7.5) +server [x86_64-darwin21]
  - rubocop-rails 2.15.2
```

**Comment by KessaPassa** [comment]
I think @tristil said right.
Actual behavior is not better.

`hash type` is not defined. so, `hash type` is using `else`.
I'll try to fix it to create a `when :hash`.
https://github.com/rubocop/rubocop/blob/4b4c2a409b14908c4344995720e04a0c65dc55e4/lib/rubocop/cop/lint/literal_in_interpolation.rb#L62-L79

**Comment by piroor** [comment]
I agree to @KessaPassa, because linter's autocorrect should not break compatibility of the target code. On this case it is expected that rubocop corrects the code exactly same as the result Ruby evaluates.
